src/api/__tests__/hooks.unit.test.ts.tsx

```diff
@@ -1,533 +1,315 @@
 import "core-js";
 import mockAxios from "jest-mock-axios";
-import * as h from "../hooks";
 import { renderHook, act } from "@testing-library/react-hooks";
-import { RequestState } from "../../shared/types";
-import { CSRF_COOKIE } from "../../auth/Auth";
-import { testUser, makeTestError } from "../../shared/test-utils";
 
+import { retrieveFactory, patchFactory } from "../api";
+import { useAPI, RequestState, useAPILazy, useAPILazyCSRF } from "../hooks";
+import {
+  MOCK_RESPONSE,
+  MOCK_ERROR,
+  MOCK_CSRF,
+  MOCK_BODY,
+  testUser,
+} from "../../shared/test-utils";
 import { AuthProvider } from "../../auth/AuthProvider";
-import ERRORS from "../../shared/errors";
+import { APIResponse } from "../../shared/types";
+import { CSRF_COOKIE } from "../../auth/Auth";
 
-describe("useApiGet", () => {
+describe("useAPI", () => {
```

Comment:
Missing test cases with getFactory

Reply:
Again, I'd lean towards not needing these since the actual source of the function call doesn't really affect what the hook does, and the only thing that matters from the hook's perspective is that the function fits the signature required.